packages/fx-core/src/component/resource/botService/retryHandler.ts

```diff
@@ -12,8 +15,9 @@ export class RetryHandler {
       retries = retries - 1;
       try {
         return await fn();
-      } catch (e) {
-        if (retries <= 0) {
+      } catch (e: any) {
+        // Directly throw 404 error, keep trying for other status code e.g. 503 400 500
+        if (retries <= 0 || [401, 403, 404, 429].includes(e.response?.status)) {
```

Comment:
It is better to make the list of [401, 403, 404, 429] as a constant which can have a meaningful name.

Reply:
Agree.